haskellPackages: Stackage 23.19 -> 23.21

sternenseemann opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `staging`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `staging` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

More information about Haskell packages in nixpkgs can be found [in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#haskell).

---

This is the follow-up to #401105. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

wolfgangwalther committed:
haskell-modules/configuration-hackage2nix: set up keep-sorted for main.yaml

We have keep-sorted as a CI job for a while, so we can make use of it.

wolfgangwalther committed:
haskell-modules/configuration-hackage2nix: run keep-sorted on main.yaml

wolfgangwalther committed:
maintainers/scripts/haskell: sort case- and locale-insensitively consistently

sort -i actually means "ignore nonprinting", not "ignore case". Also, we
need to set LC_ALL to get proper sorting for "-".

wolfgangwalther committed:
haskell-modules/configuration-common: clean up some comments

Some outdated comments removed, otherwise moved above their related
line.

sternenseemann committed:
haskell.compiler.ghcHEAD: 9.13.20250205 -> 9.13.20250428

alexfmpe committed:
haskellPackages: remove unneeded jailbreaks

sternenseemann committed:
haskellPackages.cborg: drop obsolete override

maralorn committed:
haskellPackages: remove unneeded jailbreaks (#403491)

nixpkgs-ci[bot] committed:
Merge 429c338e1585c6f7af1076c3cd3825b2a4a2e1a3 into haskell-updates

nixpkgs-ci[bot] committed:
Merge e830669088e3eb3c05cca2b80e02e04e1b2e1860 into haskell-updates

wolfgangwalther committed:
haskell-modules/configuration-{common,nix}: add comment about arbitrary insert position to avoid merge conflicts

nixpkgs-ci[bot] committed:
Merge 61f968627eaba23f587fb9166df3fe5d50f4132c into haskell-updates

sternenseemann committed:
all-cabal-hashes: 2025-04-21T04:38:52Z -> 2025-05-05T12:06:43Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

sternenseemann committed:
haskellPackages: stackage LTS 23.19 -> LTS 23.21

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

sternenseemann committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

sternenseemann committed:
Merge commit 17a47125fac into haskell-updates

wolfgangwalther committed:
haskell.packages.ghc910.ghc-lib-parser: 9.10.1.20250103 -> 9.10.2.20250503

Fix eval

wolfgangwalther committed:
haskell-modules: keep things sorted (#401492)

wolfgangwalther committed:
haskell.packages.ghc912.ghc-lib-parser: 9.12.2.20250320 -> 9.12.2.20250421

Fix eval

nixpkgs-ci[bot] committed:
Merge f309d7115b574c9f6033230678d909725293bbf6 into haskell-updates

wolfgangwalther committed:
haskellPackages: unbreak various packages

This was generated with scripts/unbreak.nu.

wolfgangwalther committed:
haskellPackages.jsonpatch: remove override

nixpkgs-ci[bot] committed:
Merge b3582c75c7f21ce0b429898980eddbbf05c68e55 into haskell-updates

nixpkgs-ci[bot] committed:
Merge dda3dcd3fe03e991015e9a74b22d35950f264a54 into haskell-updates

nixpkgs-ci[bot] committed:
Merge d53a0ca3e848cc7c7952ddc3ada1dd8f578e928d into haskell-updates

sternenseemann committed:
spacecookie: drop obsolete override

sternenseemann commented:
- [jsaddle-warp](https://hydra.nixos.org/build/296520614)/jsaddle-hello @alexfmpe @maralorn

sternenseemann committed:
haskellPackages.ghcide: allow building with hie-bios >= 0.15

sternenseemann committed:
haskell.packages.ghc912.cabal-install: pass in Cabal 3.14.2.0

No version of GHC currently ships Cabal > 3.14.1.

sternenseemann committed:
haskellPackages.stm-queue: skip flaky test case

maralorn committed:
[haskell-updates] haskellPackages.ghcide: allow building with hie-bios >= 0.15 (#405536)

alexfmpe committed:
haskellPackages.jsaddle-warp: fix installPhase

maralorn committed:
haskellPackages.jsaddle-warp: fix installPhase (#405595)

sternenseemann committed:
haskell.packages.ghc910.ghc-lib: 9.10.1.20250103 -> 9.10.2.20250503

Fix eval

sternenseemann committed:
haskell.packages.ghc912.ghc-lib: 9.12.2.20250320 -> 9.12.2.20250421

Fix eval

sternenseemann committed:
Revert "haskell-modules/configuration-hackage2nix: run keep-sorted on main.yaml"

This reverts commit 63b7dbb9d56d326bba0d07c9b660b4fac54507bd.

Reason for revert: one “sorted part” that was declared included both
supported-systems and unsupported-systems, causing elements to be
shuffled between those and causing unintended changes.

sternenseemann committed:
configuration-hackage2nix: add missing keep-sorted annotations

sternenseemann committed:
haskell-modules/configuration-hackage2nix: run keep-sorted

I've confirmed that this doesn't change anything, this time, by
regenerating the package set.

nixpkgs-ci[bot] committed:
Merge 0cc343a5bdf48c3ea66fe4c4383c643ac38f4311 into haskell-updates

rnhmjoj committed:
haskellPackages.selda-sqlite: unbreak

nixpkgs-ci[bot] committed:
Merge 1d8fe405469bc32a6c205d17b02a4a07afab85e4 into haskell-updates

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/packaging-prebuilt-haskell-binaries/64031/15

nixpkgs-ci[bot] committed:
Merge e09a2cada9c99f65baeedbfc77d0ebe1d525205c into haskell-updates

alexfmpe committed:
haskellPackages: remove patches merged upstream

maralorn committed:
haskellPackages: remove patches merged upstream (#406446)

nixpkgs-ci[bot] committed:
Merge 740ccecccb097c71993a3ad1090c9151272b3c7d into haskell-updates

nixpkgs-ci[bot] committed:
Merge 6983909b461cc0c5ddac208b0132f01c1f7d3a1d into haskell-updates

sternenseemann committed:
haskellPackages.servant-routes: patch bug to fix test suite

sternenseemann commented:
Waiting on https://github.com/symtegration/symtegration/issues/137 to unmark symtegration as broken.

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1815231](https://hydra.nixos.org/eval/1815231) of nixpkgs commit [cf12797](https://github.com/NixOS/nixpkgs/commits/cf127972bbf111593f302e81ef3a9778da162fc4) as of 2025-05-14 11:51 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

sternenseemann commented:
I've tested all remaining queued builds on Hydra which compile successfully.

@wolfgangwalther a6fc1b0875b79c2075f4a608fe1990178c2edda5 looks to alter the sort order in some files, is that expected?

sternenseemann committed:
haskell.packages.ghc9102.ghc-lib*: allow ghc-prim == 0.12.0

sternenseemann committed:
haskell.packages.ghc9102.ghcide: patch for changed ghci interface

wolfgangwalther committed:
haskellPackages.postgres-websockets: fix at run-time

Reason for the failure is still unclear and it was not possible to
reproduce outside nixpkgs, yet.

https://github.com/diogob/postgres-websockets/pull/101#issuecomment-2881016449

nixpkgs-ci[bot] committed:
Merge 234f1c87f85e4a40d23bfed1b4a6297ab7699ccc into haskell-updates

sternenseemann committed:
haskellPackages.selda-sqlite: regenerate package expression

leona-ya committed:
Merge remote-tracking branch 'origin/staging' into haskell-updates

wolfgangwalther committed:
haskellPackages.postgres-websockets: fix at run-time (#407098)

sternenseemann commented:
@leona-ya Any specific reason you merged staging into haskell-updates? We intentionally don't do that usually since it means that haskell-updates becomes a second staging-next as we loose all cache sharing with master. For this merge it did not matter since we had already finished testing it.

For the next one, though, we effectively need to wait on the next staging-next iteration (which may be a while with branch-off, I guess?) in order to not overly burden Hydra. Unfortunately reverting merges also doesn't really work for our branch (afaik at least).

leona-ya commented:
wow. i wasn't avake enough this morning. I'm sorry for that. I did this because https://github.com/NixOS/nixpkgs/actions/runs/15057762545/job/42327220097 said `master staging -> haskell-updates`. I misread this as just `staging -> haskell-updates` and didn't think enough. I'm very sorry for this.

sternenseemann commented:
I also won't complain about getting a break :) The step name sacrificed clarity for brevity, I fear.
